### PR TITLE
feature/PEER-378 : split roles, create action to add/revoke role

### PR DESCRIPTION
**Pull request by Dmitrijg24k**

### CODE REVIEW

File: app/containers/Administration/Content.tsx [hunk 1]
```diff
@@ -105,38 +129,40 @@ export const Content: React.FC<ContentProps> = ({
...
+                let roleName;
+                if (role === Roles.communityModerator) {
+                  roleName =
+                    translationMessages[locale][messages.communityModerator.id];
+                } else if (role === Roles.communityAdmin) {
+                  roleName =
+                    translationMessages[locale][
+                      messages.communityAdministrator.id
+                    ];
+                }
```
Comment (Lampardks) [inline app/containers/Administration/Content.tsx:143]:
```
let roleName = translationMessages[locale][messages.communityModerator.id];

if (role === Roles.communityAdmin) {
    roleName = translationMessages[locale][messages.communityAdministrator.id];
}
```

File: app/containers/Administration/AddRoleForm.tsx [hunk 2]
```diff
@@ -78,13 +92,29 @@ const AddModeratorForm: React.FC<AddModeratorFormProps> = ({
             `}
             className="tc"
           >
-            <FormattedMessage id={messages.addModerator.id} />
+            <FormattedMessage id={messages.addRole.id} />
           </h5>
 
-          <form onSubmit={handleSubmit(addModeratorMethod)}>
+          <form onSubmit={handleSubmit(addRoleMethod)}>
+            <Field
```
[the remark this answers is not in the document]
Comment (Dmitrijg24k) [reply app/containers/Administration/AddRoleForm.tsx]: Done

File: app/containers/Administration/Content.tsx [hunk 3]
```diff
@@ -23,53 +25,75 @@ import {
...
+  moderatorsLoading,
+  revokeRoleLoading,
+}): JSX.Element | null => {
   const moderatorRole = getCommunityRole(COMMUNITY_MODERATOR_ROLE, communityId);
   const adminRole = getCommunityRole(COMMUNITY_ADMIN_ROLE, communityId);
 
+  const usersModerator = [
+    ...new Set(moderators.map((moderator) => moderator.user)),
+  ];
+  const usersModeratorByRols = usersModerator.map((user) => {
```
[the remark this answers is not in the document]
Comment (Dmitrijg24k) [reply app/containers/Administration/Content.tsx]: Done

File: app/containers/Administration/Content.tsx (hunk 1 above)
Comment (Dmitrijg24k) [reply app/containers/Administration/Content.tsx:127]: Done in fix 1

**Review by Lampardks — APPROVED** [review]

**Pull request closed without merging**